src/ncp-spinel/SpinelNCPTaskForm.cpp

```diff
@@ -155,16 +159,37 @@ nl::wpantund::SpinelNCPTaskForm::vprocess_event(int event, va_list args)
 	// to execute.
 	EH_WAIT_UNTIL(EVENT_STARTING_TASK != event);
 
-	mLastState = mInstance->get_ncp_state();
-	mInstance->change_ncp_state(ASSOCIATING);
+	if (mUseModernBehavior) {
+		// Turn off PROP_NET_STACK_UP
+		mNextCommand = SpinelPackData(
+			SPINEL_FRAME_PACK_CMD_PROP_VALUE_SET(SPINEL_DATATYPE_BOOL_S),
+			SPINEL_PROP_NET_STACK_UP,
+			false
+		);
+		EH_SPAWN(&mSubPT, vprocess_send_command(event, args));
+		ret = mNextCommandRet;
+		require_noerr(ret, on_error);
+
+		// Turn off PROP_NET_IF_UP
+		mNextCommand = SpinelPackData(
+			SPINEL_FRAME_PACK_CMD_PROP_VALUE_SET(SPINEL_DATATYPE_BOOL_S),
+			SPINEL_PROP_NET_IF_UP,
+			false
+		);
+		EH_SPAWN(&mSubPT, vprocess_send_command(event, args));
+		ret = mNextCommandRet;
+		require_noerr(ret, on_error);
```

Comment:
The new modern behavior seem to allow a `Form` to go through even if we are already part of a network.

We may need to be careful about info from previous network we were part of (e.g., if I remember correctly , OT may not clear network data on IF/Stack down)...